### PR TITLE
Fix TypeScript import path

### PR DESCRIPTION
**Pull request by lachenmayer**
Hi there, the TypeScript path in `package.json` is incorrect, causing the import to fail with `Could not find a declaration file for module 'react-native-restart'.` when used from TypeScript.

I've previously had to make this exact fix this in another library too (https://github.com/greentriangle/react-native-leveldb/pull/1), so I believe this is probably an issue with `react-native-builder-bob`. For now, this fix should solve the import issue in this library though. I will open an issue on the `react-native-builder-bob` repo too.

Thanks a lot!

### CODE REVIEW

**Comment by SaeedZhiany** [comment]
@avishayil 

I have this problem too.
please review and merge this, Thanks

File: package.json
Comment (SaeedZhiany) [inline package.json:7]: @lachenmayer 

there is not a path for `lib/typescript/index.d.ts` in this repository anymore. are you sure about that?

**Comment by SaeedZhiany** [comment]
@avishayil 

I have this problem too.
please review and merge this if it's fix the problem, Thanks

**Comment by SaeedZhiany** [comment]
@lachenmayer 

I applied your changes manually in my project, but I still, have a TS error.

image


is it working perfectly on your project?

**Comment by avishayil** [comment]
I'll review it this week, sorry for the time it takes...

**Comment by Adnan-Bacic** [comment]
im getting the same error but with JS instead of TS. do you know if its the same problem?

**Comment by SaeedZhiany** [comment]
@avishayil is there any progress on this? Thanks for your time.

**Comment by SaeedZhiany** [comment]
@avishayil is there any progress on this? Thanks for your time.

**Comment by TheWirv** [comment]
I encountered the same problem, and @lachenmayer's fix is an easy one that works. :) I wanted second the notion to approve this PR because this is an issue that we cannot `patch-package` unfortunately, due to it concerning the `package.json` file. :/

**Comment by SaeedZhiany** [comment]
@TheWirv 

have you encountered the problem I mentioned in [this comment](https://github.com/avishayil/react-native-restart/pull/160#issuecomment-812825359) after applying @lachenmayer's fix?

**Comment by TheWirv** [comment]
> @TheWirv
> 
> have you encountered the problem I mentioned in [this comment](https://github.com/avishayil/react-native-restart/pull/160#issuecomment-812825359) after applying @lachenmayer's fix?

@SaeedZhiany No, it works perfectly fine for me, sorry. :(

**Comment by SaeedZhiany** [comment]
What typescript version you are using?

**Comment by TheWirv** [comment]
> 
> 
> What typescript version you are using?

4.2.4

**Comment by SaeedZhiany** [comment]
I tried that version, it seems the problem is not related to the typescript version. However, [the update section in this comment](https://github.com/avishayil/react-native-restart/pull/160#issuecomment-812825359) is also resolving my problem, I hope @avishayil review and fix the issue soon.

**Comment by SaeedZhiany** [comment]
I don't know why but I just figured out that the structure of the repository is pretty different from what we are installing.

| installed package |  sadsa|
|--------------------|--------|
|image|image|

there is no `lib` directory in the repository and I'm wondering how it has been generated.

However, I found a way to resolve the TS error I reported in [this comment](https://github.com/avishayil/react-native-restart/pull/160#issuecomment-812825359). when I replace the content of the file `/lib/typescript/index.d.ts` with the code below, the problem is resolving.

```tsx
declare module 'react-native-restart' {
    type RestartType = {
        Restart(): void;
    };
    const RNRestart: RestartType;
    export default RNRestart;
}
```

I wanted to make a PR for this to integrate @lachenmayer PR and my solution to resolve typing problems but as I said early, I couldn't the file in the repository to modify.

@avishayil please apply these changes soon, I really need them in my project. Thanks

**Review by avishayil — APPROVED** [review]

**Comment by SaeedZhiany** [comment]
@avishayil did you read my comments above?

**Comment by avishayil** [comment]
Hi @SaeedZhiany the lib folder is create using the `bob build` command which packages and prepares the artifact to be deployed to npm.

**Comment by a-eid** [comment]
hey wondering if you will be making a release with this fix soon ?

**Comment by xiankuncheng** [comment]
Thank you all for the effort to solve this issue for the community.
I've met the same issue this week as well, and figured out it's been fixed already, that's amazing.

Do we have any plan to release a new version to npm soon? Since the latest version on npm (0.0.22) is still with the wrong type path in the `package.json` file.
Appreciate all the help 🙏

**Comment by badsyntax** [comment]
@avishayil can we get a new release please?

**Comment by badsyntax** [comment]
While we wait for a release i'm patching this issue by creating a new file called `react-native-restart.d.ts` with the following contents:

```ts
declare module 'react-native-restart' {
  type RestartType = {
    Restart(): void;
  };
  const _default: RestartType;
  export default _default;
}
```

**Comment by avishayil** [comment]
Sure. I'm waiting for #167 to release a new version with Windows support.

**Comment by barbieri** [comment]
would you mind a new release? 0.0.22 is still wrong :-/

**Comment by crutchcorn** [comment]
+1 to @barbieri's comment. `0.0.22` uses the wrong import path, despite the code in the repo being correct

**Comment by vonovak** [comment]
@avishayil would you be so kind and release the new version? There is no need to wait for anything; thank you! :)

**Comment by angelos3lex** [comment]
@avishayil Can you please release master on npm ? This is blocking us from updating the lib, as its was not a prob back in 0.0.17. Would love to avoid suggested workarounds too, since this is fixed already :)

**Comment by pedrohff360** [comment]
Do we have some expectation about the next release? :)

**Comment by bombillazo** [comment]
Following up to see when this will get release.

**Pull request merged**
